### ISSUE TITLE
Lambda Event Source Mapping From Kinesis Does Not Always Honor Batch Size

### ISSUE THREAD

**Issue by tmoser2525** [localstack/localstack#1131]
I have a setup where there is a node.js application that writes to a kinesis stream using putRecords for two items. The kinesis stream feeds a Lambda function that is configured with an eventSourceMapping with a batch size of 1. However, when I debug what the Records Kinesis feeds the lambda for an invocation, I see that the lambda is processing two records. Both of these records are on the same shard and have a sequence number which is pretty close.

The interesting thing is that if I switch the node.js published to use putRecord instead of putRecords, the kinesis records have a slightly farther apart sequence number and the lambda is fed the records individually.

The java code to configure the event source mapping
```
        AWSLambdaAsync lambdaClient = AWSResourceFactory.getLambdaClient();
        CreateEventSourceMappingRequest request = new CreateEventSourceMappingRequest();
        request.setEventSourceArn(kinesisArn);
        request.setFunctionName(functionName);
        request.setEnabled(true);
        request.setStartingPosition(EventSourcePosition.TRIM_HORIZON);
        request.setBatchSize(1);
        lambdaClient.createEventSourceMapping(request);
```

The localstack lambda output that shows multiple records being read at a time:

```
{
	"Records": [{
		"eventID": "shardId-000000000000:49593014281924840790139112603493819506073986345293840386",
		"eventSourceARN": "arn:aws:kinesis:us-east-1:000000000000:stream/MyStream",
		"kinesis": {
			"partitionKey": "ed0f756b97234f4fae9e3d1a886b8fbc",
			"data": "myData1",
			"sequenceNumber": "49593014281924840790139112603493819506073986345293840386"
		}
	}, {
		"eventID": "shardId-000000000000:49593014281924840790139112603495028431893600974468546562",
		"eventSourceARN": "arn:aws:kinesis:us-east-1:000000000000:stream/MyStream",
		"kinesis": {
			"partitionKey": "abee63a9299385699d65e89a290a9065",
			"data": "myData2",
			"sequenceNumber": "49593014281924840790139112603495028431893600974468546562"
		}
	}]
}
```

**Comment by whummer** [comment]
Hi, thanks for reporting @tmoser2525 . This should be relatively easy to achieve - instead of creating a single `event` with all records, we could create multiple events according to the batch size configuration of the event source mapping here:

https://github.com/localstack/localstack/blob/master/localstack/services/awslambda/lambda_api.py#L210-L219

Would you be able to create a pull request and contribute a simple integration test? That would be great! Thanks

**Comment by whummer** [comment]
@tmoser2525 This should be fixed in #2110 . Can you please give it a try with the `latest` version? Please report here if the problem persists. Thanks
